File: sevenkyu/recursivereversestring/src/main/java/hu/crs/codewars/recursivereversestring/Kata.java
package hu.crs.codewars.recursivereversestring;

public class Kata {
    public String reverse(String str) {
        if (str.length() == 1) {
            return str;
        } else {
            return reverse(str.substring(1)) + str.charAt(0);
        }
    }
}

File: src/main/kotlin/hu/crs/codewars/sevenkyu/vowelcount/Kata.kt
package hu.crs.codewars.sevenkyu.vowelcount

/**
 * Vowel Count
 *
 * https://www.codewars.com/kata/54ff3102c1bad923760001f3
 */
fun getCount(str: String): Int {
    return str.chars()
            .mapToObj { n -> n.toChar() }
            .filter { n -> n == 'a' || n == 'e' || n == 'i' || n == 'o' || n == 'u' }
            .count()
            .toInt()
}
